core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java

```diff
@@ -27,40 +27,64 @@
 import org.elasticsearch.search.aggregations.AggregatorFactories.Builder;
 import org.elasticsearch.search.aggregations.AggregatorFactory;
 import org.elasticsearch.search.aggregations.support.AggregationContext;
+import org.elasticsearch.search.aggregations.support.ValueType;
 import org.elasticsearch.search.aggregations.support.ValuesSource.Numeric;
+import org.elasticsearch.search.aggregations.support.ValuesSourceAggregatorFactory;
 import org.elasticsearch.search.aggregations.support.ValuesSourceConfig;
+import org.elasticsearch.search.aggregations.support.ValuesSourceType;
 
 import java.io.IOException;
 import java.util.Objects;
 
 public class DateHistogramAggregationBuilder extends AbstractHistogramBuilder<DateHistogramAggregationBuilder> {
-
     public static final String NAME = InternalDateHistogram.TYPE.name();
     public static final ParseField AGGREGATION_NAME_FIELD = new ParseField(NAME);
 
+    private long interval;
     private DateHistogramInterval dateHistogramInterval;
+    private long offset = 0;
+    private ExtendedBounds extendedBounds;
 
     public DateHistogramAggregationBuilder(String name) {
-        super(name, InternalDateHistogram.HISTOGRAM_FACTORY);
+        super(name, InternalDateHistogram.TYPE, ValuesSourceType.NUMERIC, ValueType.DATE);
     }
 
-    /**
-     * Read from a stream.
-     */
     public DateHistogramAggregationBuilder(StreamInput in) throws IOException {
-        super(in, InternalDateHistogram.HISTOGRAM_FACTORY);
+        super(in, InternalDateHistogram.TYPE, ValuesSourceType.NUMERIC, ValueType.DATE);
+        interval = in.readLong();
         dateHistogramInterval = in.readOptionalWriteable(DateHistogramInterval::new);
+        offset = in.readLong();
+        if (in.readBoolean()) {
```

Comment:
`readOptionalWriteable`? I love making these methods shorter so they are easier to scan visually.